kernel/tests/golden_tables.rs

```diff
@@ -122,29 +125,34 @@ fn assert_schema_fields_match(schema: &Schema, golden: &Schema) {
     }
 }
 
-// copied from DAT
-// some things are equivalent, but don't show up as equivalent for `==`, so we normalize here
-fn normalize_col(col: Arc<dyn Array>) -> Arc<dyn Array> {
-    if let DataType::Timestamp(unit, Some(zone)) = col.data_type() {
-        if **zone == *"+00:00" {
-            arrow_cast::cast::cast(&col, &DataType::Timestamp(*unit, Some("UTC".into())))
-                .expect("Could not cast to UTC")
-        } else {
-            col
+fn assert_cols_eq(actual: &dyn Array, expected: &dyn Array) {
+    match actual.data_type() {
+        DataType::Struct(_) => {
+            let actual_sa = actual.as_struct();
+            let expected_sa = expected.as_struct();
+            assert_eq(actual_sa, expected_sa);
+        }
+        DataType::List(_) => {
+            let actual_la = actual.as_list::<i32>();
+            let expected_la = expected.as_list::<i32>();
+            assert_cols_eq(actual_la.values(), expected_la.values());
+        }
+        DataType::Map(_, _) => {
+            let actual_ma = actual.as_map();
+            let expected_ma = expected.as_map();
+            assert_cols_eq(actual_ma.keys(), expected_ma.keys());
+            assert_cols_eq(actual_ma.values(), expected_ma.values());
+        }
+        _ => {
+            assert_eq!(actual, expected, "Column data didn't match.");
         }
-    } else {
-        col
     }
 }
 
-// copied from DAT
-fn assert_columns_match(actual: &[Arc<dyn Array>], expected: &[Arc<dyn Array>]) {
-    for (actual, expected) in actual.iter().zip(expected) {
-        let actual = normalize_col(actual.clone());
-        let expected = normalize_col(expected.clone());
-        // note that array equality includes data_type equality
-        // See: https://arrow.apache.org/rust/arrow_data/equal/fn.equal.html
-        assert_eq!(&actual, &expected, "Column data didn't match.");
+fn assert_eq(actual: &StructArray, expected: &StructArray) {
+    assert_fields_match(actual.fields().iter(), expected.fields().iter());
+    for (actual_col, expected_col) in actual.columns().iter().zip(expected.columns()) {
```

Comment:
Lots of zipping here... we need to validate counts as well, no?

Reply:
yep, good catch, fixed.